src/state_machine/transitions.cpp

```diff
@@ -0,0 +1,177 @@
+    log.ERR("STM", "Critical failure in batteries");
+    return true;
+  }
+  return false;
+}
+
+State *checkEmergencyStationery(Logger &log, EmergencyBrakes embrakes_data, Navigation nav_data,
+                                Batteries batteries_data, Telemetry telemetry_data,
+                                Sensors sensors_data, Motors motors_data)
+{
+  if (!checkEmergencyGeneral(log, embrakes_data, nav_data, batteries_data, telemetry_data,
+                             sensors_data, motors_data)) {
+    return NULL;
+  }
+  return FailureStopped::instance_;
+}
+
+State *checkEmergencyMoving(Logger &log, EmergencyBrakes embrakes_data, Navigation nav_data,
+                            Batteries batteries_data, Telemetry telemetry_data,
+                            Sensors sensors_data, Motors motors_data)
+{
+  if (!checkEmergencyGeneral(log, embrakes_data, nav_data, batteries_data, telemetry_data,
+                             sensors_data, motors_data)) {
+    return NULL;
+  }
+  return FailureBraking::instance_;
+}
+
+//--------------------------------------------------------------------------------------
+// Module Status
+//--------------------------------------------------------------------------------------
+
+State *checkModulesInitialised(Logger &log, EmergencyBrakes embrakes_data, Navigation nav_data,
+                               Batteries batteries_data, Telemetry telemetry_data,
+                               Sensors sensors_data, Motors motors_data)
+{
+  if (!telemetry_data.calibrate_command) return NULL;
+
+  if (embrakes_data.module_status != ModuleStatus::kInit) return NULL;
+  if (nav_data.module_status != ModuleStatus::kInit) return NULL;
+  if (batteries_data.module_status != ModuleStatus::kInit) return NULL;
+  if (telemetry_data.module_status != ModuleStatus::kInit) return NULL;
+  if (sensors_data.module_status != ModuleStatus::kInit) return NULL;
+  if (motors_data.module_status != ModuleStatus::kInit) return NULL;
+
+  log.INFO("STM", "Calibrate command received and all modules initialised");
+  return Calibrating::instance_;
+}
+
+State *checkModulesReady(Logger &log, EmergencyBrakes embrakes_data, Navigation nav_data,
+                         Motors motors_data)
+{
+  // We're only checking Navigation, Motors and Embrakes because only those modules are doing
+  // calibration.
+  if (embrakes_data.module_status != ModuleStatus::kReady) return NULL;
+  if (nav_data.module_status != ModuleStatus::kReady) return NULL;
+  if (motors_data.module_status != ModuleStatus::kReady) return NULL;
+
+  log.INFO("STM", "All modules calibrated");
+  return Ready::instance_;
+}
+
+//--------------------------------------------------------------------------------------
+// Telemetry Commands
+//--------------------------------------------------------------------------------------
+
+State *checkLaunchCommand(Logger &log, Telemetry telemetry_data)
+{
+  if (!telemetry_data.calibrate_command) return NULL;
```

Comment:
Shouldn't it be `launch_command`? As I can't see `launch_command` anywhere else in the code

Reply:
Good spot! Will fix.